optihood/IO/readers.py

```diff
@@ -316,26 +322,24 @@ def add_electricity_cost(self, nodesData, cluster_size, time_index: _pd.Datetime
         # float64 is not an instance of float!!
         if isinstance(electricityCost, (float, int, _np.float64, _np.int64)):
             # for constant cost
-            electricityCostValue = electricityCost
             _log.info("Constant value for electricity cost")
-            nodesData["electricity_cost"] = _pd.DataFrame()
-            nodesData["electricity_cost"]["cost"] = (nodesData["demandProfiles"][1].shape[0]) * [
-                electricityCostValue]
-            nodesData["electricity_cost"].index = nodesData["demandProfiles"][1].index
+            df = _pd.DataFrame()
+            df["cost"] = (nodesData["demandProfiles"][1].shape[0]) * [electricityCost]
+            df.index = nodesData["demandProfiles"][1].index
         elif not _os.path.exists(electricityCost):
             _log.error("Error in electricity cost file path")
             raise FileNotFoundError(electricityCost)
         else:
-            nodesData["electricity_cost"] = _pd.read_csv(electricityCost, delimiter=";")
+            df = _pd.read_csv(electricityCost, delimiter=";")
             # set datetime index
-            nodesData["electricity_cost"].set_index("timestamp", inplace=True)
-            nodesData["electricity_cost"].index = _pd.to_datetime(nodesData["electricity_cost"].index,
-                                                                  format='%d.%m.%Y %H:%M')
-            nodesData["electricity_cost"] = self.clip_to_time_index(nodesData["electricity_cost"], time_index)
+            df.set_index("timestamp", inplace=True)
+            df.index = _pd.to_datetime(df.index, format='%d.%m.%Y %H:%M')
+            df = self.clip_to_time_index(df, time_index)
 
         if cluster_size:
-            electricityCost = self.cluster_and_multiply_desired_column(nodesData["electricity_cost"], cluster_size)
-            nodesData["electricity_cost"] = electricityCost
+            df = self.cluster_and_multiply_desired_column(df, cluster_size)
+
+        nodesData["electricity_cost"] = df
```

Comment:
Redundant code for preparing df for electricity cost, impact, natural gas cost & impact. It would be cleaner to add this logic to a separate function.